presto-cache/src/main/java/com/facebook/presto/cache/filemerge/FileMergeCacheManager.java

```diff
@@ -153,8 +153,13 @@ public FileMergeCacheManager(
 
         this.cacheSizeCalculateExecutor.scheduleAtFixedRate(
                 () -> {
-                    cacheScopeFiles.keySet().forEach(cacheIdentifier -> cacheScopeSizeInBytes.put(cacheIdentifier, getCacheScopeSizeInBytes(cacheIdentifier)));
-                    cacheScopeSizeInBytes.keySet().removeIf(key -> !cacheScopeFiles.containsKey(key));
+                    try {
+                        cacheScopeFiles.keySet().forEach(cacheIdentifier -> cacheScopeSizeInBytes.put(cacheIdentifier, getCacheScopeSizeInBytes(cacheIdentifier)));
+                        cacheScopeSizeInBytes.keySet().removeIf(key -> !cacheScopeFiles.containsKey(key));
+                    }
+                    catch (Throwable t) {
+                        log.error(t, "Error calculating cache size");
```

Comment:
In my opinion we should log and throw until and unless we are confident it is OK to retry and also that retry can fix things. If we get into a situation that 1 eentry is causing problems and that is causing this executor to always exit midway, then dying and complaining is probably a better solution.

Reply:
It looks like it is pretty consistent in our codebase to log and let it retry. Do you think we should stop retrying for this specific case? Or in general?

Reply:
In this case and in general for most cases until and unless the underlying logic involves RPC calls. Here this thread is modifying class members and if the error is non-recoverable, then it is possible that we get half of the state in the variable and we think that things are working - so better screw it up well

Reply:
Currently the code doesn't have any RPC calls, but in theory it might at some point. Logging and retrying is the standard way of handling problems in scheduled tasks. Further, when we build alerting, we are going to be able to detect them quickly. It feels like killing the JVM is too drastic measure, and it might be potentially unsafe as somebody might not realize how the error is handled and add some code that is potentially recoverable.

Thoughts?